apps/prairielearn/assets/scripts/navbarClient.ts

```diff
@@ -0,0 +1,154 @@
+    Cookies.set('pl_requested_course_instance_role', 'None', { path: '/', expires: expires });
+    Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+    if (viewType === 'student') {
+      location.reload();
+    }
+  });
+
+  document.querySelector('#navbar-user-view-instructor')?.addEventListener('click', () => {
+    Cookies.remove('pl_requested_course_role', { path: '/' });
+    Cookies.remove('pl_requested_course_instance_role', { path: '/' });
+    Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+    if (viewType === 'instructor') {
+      location.reload();
+    }
+  });
+
+  document.querySelector('#navbar-user-view-student-no-rules')?.addEventListener('click', () => {
+    Cookies.remove('pl_requested_course_role', { path: '/' });
+    Cookies.remove('pl_requested_course_instance_role', { path: '/' });
+    Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+    if (viewType === 'student') {
+      location.reload();
+    }
+  });
+
+  document.querySelector('#navbar-user-view-student')?.addEventListener('click', () => {
+    const courseRole = hasInstructorAccess ? 'None' : authnCourseRole;
+    const courseInstanceRole = hasInstructorAccess ? 'None' : authnCourseInstanceRole;
+
+    Cookies.set('pl_requested_course_role', courseRole, { path: '/', expires: expires });
+    Cookies.set('pl_requested_course_instance_role', courseInstanceRole, {
+      path: '/',
+      expires: expires,
+    });
+    Cookies.set('pl_requested_data_changed', 'true', { path: '/' });
+
+    if (viewType === 'student') {
+      location.reload();
+    }
+  });
+
+  const navbarUserEffectiveUidInput = document.querySelector<HTMLInputElement>(
+    '#navbar-user-effective-uid-input',
+  );
+  const navbarUserEffectiveUidButton = document.querySelector<HTMLButtonElement>(
+    '#navbar-user-effective-uid-button',
+  );
+
+  navbarUserEffectiveUidInput?.addEventListener('keyup', function (e) {
+    if (navbarUserEffectiveUidInput.value.trim() !== '') {
+      navbarUserEffectiveUidButton.removeAttribute('disabled');
+    } else {
+      navbarUserEffectiveUidButton.setAttribute('disabled', 'true');
+    }
+  });
+
+  navbarUserEffectiveUidInput?.addEventListener('keypress', function (e) {
+    if (e.key === 'Enter') {
+      const uid = navbarUserEffectiveUidInput.value.trim();
+      if (uid !== '') {
+        document.cookie = `pl_requested_uid=${uid}; path=/; expires=${expires}`;
+        document.cookie = 'pl_requested_data_changed; path=/';
+        location.reload();
+      }
+    }
+  });
+
+  navbarUserEffectiveUidButton?.addEventListener('click', function () {
+    const uid = navbarUserEffectiveUidInput.value.trim();
+    if (uid !== '') {
+      document.cookie = `pl_requested_uid=${uid}; path=/; expires=${expires}`;
+      document.cookie = 'pl_requested_data_changed; path=/';
+      location.reload();
+    }
+  });
```

Comment:
I'm pretty sure I know the answer to this, but: is there any good reason why we're using `document.cookie` directly here instead of `js-cookie`?

Reply:
The original code isn't, are you sure you didn't convert something inadvertently?

https://github.com/PrairieLearn/PrairieLearn/blob/800ec7f1f550ac8dc8934d46bab4744238bd6e0f/apps/prairielearn/src/pages/partials/navbar.ejs#L494-L501

Reply:
Man, that's what I get for having Copilot rewrite this for me! 😓

Reply:
Changed in 5899a02.